epilogue-processor/src/main/java/edu/wpi/first/epilogue/processor/LoggerGenerator.java

```diff
@@ -220,35 +239,75 @@ private void writeLoggerFile(
           out.println("  private static final VarHandle $" + varHandleField.getSimpleName() + ";");
         }
         out.println();
+      }
 
-        var classReference = simpleClassName + ".class";
+      if (requiresReflection) {
+        for (var reflectionField : reflectionFields) {
+          // This field needs reflection to access.
+          // Cache it in the class to avoid lookups
+          out.println("  private static final Field $" + reflectionField.getSimpleName() + ";");
+        }
+        out.println();
+      }
 
+      // Static initializer block to load VarHandles and reflection fields
+      if (requiresVarHandles || requiresReflection) {
         out.println("  static {");
-        out.println("    try {");
-        out.println(
-            "      var lookup = MethodHandles.privateLookupIn("
-                + classReference
-                + ", MethodHandles.lookup());");
 
-        for (var varHandleField : varHandleFields) {
-          var fieldName = varHandleField.getSimpleName();
+        if (requiresVarHandles) {
+          out.println("    try {");
+
+          var classReference = simpleClassName + ".class";
           out.println(
-              "      $"
-                  + fieldName
-                  + " = lookup.findVarHandle("
+              "      var lookup = MethodHandles.privateLookupIn("
                   + classReference
-                  + ", \""
-                  + fieldName
-                  + "\", "
-                  + m_processingEnv.getTypeUtils().erasure(varHandleField.asType())
-                  + ".class);");
+                  + ", MethodHandles.lookup());");
```

Comment:
We can use a separate private lookup for each type and generate something like this:

```java
// given
@Logged class Superclass {
  private double superclassVar;
}

@Logged class Subclass extends Superclass {
  private int subclassVar;
}

// generate
var rootLookup = MethodHandles.lookup();
var subclassLookup = MethodHandles.privateLookupIn(Subclass.class, rootLookup);
var superclassLookup = MethodHandles.privateLookupIn(Superclass.class, rootLookup);

$subclassVar = subclassLookup.findVarHandle(Subclass.class, "subclassVar", int.class);
$superclassVar = superclassLookup.findVarHandle(Superclass.class, "superclassVar", double.class);
```

Then there's no need for runtime reflection